modules/core/src/main/java/org/apache/fluo/core/impl/TransactionImpl.java

```diff
@@ -265,6 +269,42 @@ public Bytes get(Bytes row, Column column) {
     return ret;
   }
 
+  @Override
+  public CompletableFuture<Bytes> getAsync(Bytes row, Column column) {
+    if (asyncReader == null) {
+      asyncReader = new AsyncReader(this);
+    }
+
+    return asyncReader.get(row, column);
+  }
+
+  @Override
+  public CompletableFuture<Bytes> getAsync(Bytes row, Column column, Bytes defaultValue) {
+    if (asyncReader == null) {
+      asyncReader = new AsyncReader(this);
+    }
+
+    return asyncReader.get(row, column, defaultValue);
+  }
+
+  @Override
+  public CompletableFuture<String> getsAsync(String row, Column column) {
+    if (asyncReader == null) {
```

Comment:
Could shorten these method with a method like the following

```java
private AsyncReader getAsyncReader() {
  if (asyncReader == null) {
￼    asyncReader = new AsyncReader(this);
￼  }
}
```

Then could do the following below

```java
  return getAsyncReader().gets(row, column);
```